<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>sort_visualizerClass</class>
 <widget class="QMainWindow" name="sort_visualizerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>402</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>600</width>
    <height>400</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>sort_visualizer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QListWidget" name="algorithms_list_widget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>150</width>
        <height>16777215</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="1" column="0" rowspan="2">
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="1">
       <widget class="QSpinBox" name="delay_spin_box">
        <property name="maximum">
         <number>2147483647</number>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="array_size_spin_box">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
        <property name="minimum">
         <number>2</number>
        </property>
        <property name="maximum">
         <number>2147483647</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>delay in nanosecons</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QPushButton" name="shuffle_button">
        <property name="text">
         <string>shuffle</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QPushButton" name="sort_button">
        <property name="text">
         <string>sort</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QPushButton" name="pause_button">
        <property name="text">
         <string>pause</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="QPushButton" name="break_button">
        <property name="text">
         <string>break</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QPushButton" name="step_button">
        <property name="text">
         <string>step</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>array size</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="1" rowspan="3">
     <widget class="array_frame" name="frame">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>600</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action1">
   <property name="text">
    <string>1</string>
   </property>
  </action>
  <action name="action_std_sort">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>std::sort</string>
   </property>
  </action>
  <action name="action_std_stable_sort">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>std::stable_sort</string>
   </property>
  </action>
  <action name="action_std_heap_sort">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>std::heap_sort</string>
   </property>
  </action>
  <action name="action_boost_pdqsort">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>boost::pdqsort</string>
   </property>
  </action>
  <action name="action_boost_integer_sort">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>boost::integer_sort</string>
   </property>
  </action>
  <action name="action_boost_pdqsort_branchless">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>boost::pdqsort_branchless</string>
   </property>
  </action>
  <action name="action_qSort">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>qSort</string>
   </property>
  </action>
  <action name="action_qStableSort">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>qStableSort</string>
   </property>
  </action>
  <action name="action_dual_pivot_quick_sort">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>dual_pivot_quick_sort</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>array_frame</class>
   <extends>QFrame</extends>
   <header>array_frame.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <slots>
  <slot>on_push_button_clicked()</slot>
 </slots>
</ui>
